lib/src/common/shared.rs

```diff
@@ -107,7 +107,7 @@ impl<T> Shared<T> {
         Box::new(RefCell::borrow_mut(&self.0))
     }
 
-    pub fn cloned(&self) -> Self where T: Clone {
+    pub fn clone_inner(&self) -> Self where T: Clone {
```

Comment:
My issue is that `cloned` is only one letter off from `Shared<T>::clone()` which clones the pointer but not the inside object.  This confusion actually bit me when I didn't spot the difference between `clone` and `cloned` and it cost me some time debugging an unintended behavior change.

In my opinion, `Option::cloned()` isn't quite the same because it takes an `Option<&T>` arg and returns an `Option<T>`, so the type system will catch you if there is a mistake.  On the other hand, both `Shared<T>::clone()` and `Shared<T>::cloned() / clone_inner()` return a new `Shared<T>` so the only difference is behavior

Reply:
Well I read `cloned` as return the instance of the wrapper/container with a cloned content. `Option<T>::cloned` thus also makes sense. I agree `cloned` and `clone` are not very distinguishable by an eye. Lets leave `clone_inner` than.